### PR TITLE
Remove static ingress credential from nginx config for VLC add-on

### PR DESCRIPTION
**Pull request by joelmoses**
Use the same generated secret as the add-on uses to seed the VLC login and tie it to the NGINX ingress. Removes the static basic credential in the nginx.conf file of the original code.

EDIT: On owner request, uses separate secret for Ingress credential.

Reference to issue here: https://github.com/home-assistant/addons/issues/1873

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @joelmoses, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;addons#1878).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by joelmoses** [comment]
Putting this to the attention of @pvizeli and @MartinHjelmare who merged the initial addon.

**Comment by pvizeli** [comment]
This need core discovery support to work, we bump the version until a core come out that this can use

**Comment by pvizeli** [comment]
Sorry read it wrong :)

File: vlc/rootfs/etc/cont-init.d/ingress.sh [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+# Create VLC secret
+# ==============================================================================
+
+# Generate password
+if bashio::fs.file_exists /data/secret; then
+    bashio::exit.ok
+else
+    pwgen 64 1 > /data/secret
+fi
+
```
Comment (pvizeli) [inline vlc/rootfs/etc/cont-init.d/ingress.sh:12]: Create own secret for the http and restore the `secret.sh` file

File: vlc/rootfs/etc/services.d/vlc/run [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+# Start C-VLC daemon and enable discovery
 # ==============================================================================
 readonly PASSWORD="$(cat /data/secret)"
 
 # Send out discovery information to Home Assistant
 ./discovery &
 
 # Run daemon
-exec cvlc -I http --extraintf telnet --http-host 127.0.0.1 --http-password "INGRESS" --telnet-password "${PASSWORD}" --aout=pulse
+exec cvlc -I http --extraintf telnet --http-host 127.0.0.1 --http-password "${PASSWORD}" --telnet-password "${PASSWORD}" --aout=pulse
```
Comment (pvizeli) [inline vlc/rootfs/etc/services.d/vlc/run:11]: load his secret for http

**Review by pvizeli — CHANGES_REQUESTED** [review]

File: vlc/rootfs/etc/cont-init.d/ingress.sh (hunk 1 above)
Comment (joelmoses) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:12]: Certainly can, but is there any harm in using the same secret given to discovery (Telnet) for ingress (HTTP) as well? Simplifies things somewhat, and neither are values that the user will ever need to know.

Comment (pvizeli) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:12]: the telnet secret is shared, it's not a lot better than the static secret, just a bit more feignedly secure. We can make an `/data/secret.ingress` and `/data/secret.integration`.

Comment (pvizeli) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:12]: You don't need cleanup the old secret file, that is not public jet

Comment (joelmoses) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:12]: I'm not sure I understand fully why it's important to use two pwgen runs for a non-user-facing secret value, but I suppose it doesn't really matter in the end.

I'll make them separate at your request.

**Comment by joelmoses** [comment]
@pvizeli - All requested changes made.

File: vlc/rootfs/etc/services.d/vlc/run [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 # ==============================================================================
-# Start Z-Wave JS service for Z-Wave radio
+# Start C-VLC daemon and enable discovery
 # ==============================================================================
-readonly PASSWORD="$(cat /data/secret)"
+readonly TELNET_PASSWORD="$(cat /data/telnet-secret)"
+readonly INGRESS_PASSWORD="$(cat /data/ingress-secret)"
 
 # Send out discovery information to Home Assistant
 ./discovery &
```
Comment (pvizeli) [inline vlc/rootfs/etc/services.d/vlc/run:9]: Need also update the discovery data

File: vlc/rootfs/etc/cont-init.d/ingress.sh [hunk 4]
```diff
@@ -1,11 +1,23 @@
 #!/usr/bin/with-contenv bashio
+# ==============================================================================
+# Create VLC secret
+# ==============================================================================
+
+# Generate password
+if bashio::fs.file_exists /data/ingress-secret; then
+    bashio::exit.ok
```
Comment (pvizeli) [inline vlc/rootfs/etc/cont-init.d/ingress.sh:8]: This means it will never generate an ingress config on the second run. This add-on can only run once

Comment (joelmoses) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:8]: I patterned it after the way you were handling the telnet-secret; is there any harm in generating an ingress secret one time and maintaining the same secret for future runs? That's how it was before when the password was a static value of ":INGRESS".

I don't mind making it generate and use a new ingress secret on every container restart, but it seems counterproductive to me.

File: vlc/rootfs/etc/services.d/vlc/run (hunk 3 above)
Comment (joelmoses) [reply vlc/rootfs/etc/services.d/vlc/run:9]: Yep, I missed that. Thanks.

Comment (joelmoses) [reply vlc/rootfs/etc/services.d/vlc/run:9]: Resolved with https://github.com/home-assistant/addons/pull/1878/commits/94f57fc15c33cf41a0a5c212906208797cde01f0

File: vlc/rootfs/etc/cont-init.d/ingress.sh (hunk 4 above)
Comment (pvizeli) [reply vlc/rootfs/etc/cont-init.d/ingress.sh:8]: What do you think will be happening with the code after this?

**Comment by pvizeli** [comment]
Looks stale, you can anytime reopen it after the comments are address and the PR is rebased

**Pull request closed without merging**